maven.compatibility/source/java/maven/compatibility/ProjectConfigurationRequestWrapper.java

```diff
@@ -0,0 +1,51 @@
+package maven.compatibility;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.m2e.core.project.configurator.ProjectConfigurationRequest;
+
+public class ProjectConfigurationRequestWrapper
+{
+    public static IProject getProject(ProjectConfigurationRequest request)
```

Comment:
I wonder if this is the way to go... 
Using reflection is not that complicate.
See here:

https://github.com/groovy/groovy-eclipse/blob/master/extras/org.codehaus.groovy.m2eclipse/src/org/codehaus/groovy/m2eclipse/GroovyProjectConfigurator.java

Reply:
I have to agree at least in part - I don't object to a new project helping with this, but it should be buildable rather than checking in the jar directly.